plugins/plugin-core-support/src/lib/cmds/history/history.ts

```diff
@@ -107,8 +107,11 @@ const showHistory = ({ argv, parsedOptions: options }) => {
   const filterStr = filterIdx > 0 && argv[filterIdx]
   const filter = filterStr ? line => line.raw.indexOf(filterStr) >= 0 : () => true
 
-  const startIdx = Math.max(0, historyModel.getCursor() - N - 1)
-  const endIdx = historyModel.getCursor() - 1
+  /* don't use historyModel.getCursor() to get the index here
+  since the cursor will be changed if the users get the history
+  command in history by key up and down events */
+  const startIdx = Math.max(0, historyModel.lines.length - N - 1)
+  const endIdx = historyModel.lines.length
```

Comment:
`endIdx` here doesn't need to `- 1` because the command & output will be added to the history model after command evaluation. (we used to do it before command evaluation, and I think that's why we have `const endIdx = historyModel.getCursor() - 1`.